src/super_gradients/modules/backbone_replacement_utils.py

```diff
@@ -0,0 +1,56 @@
+from typing import Union, Optional, Callable
+
+import torch
+from torch import nn
+
+__all__ = ["replace_input_channels_with_random_weights"]
+
+
+def compute_new_weights(
+    module: Union[nn.Conv2d, nn.Linear, nn.Module],
```

Comment:
Dont think nn.Linear belongs here.
Actually, nn.Module is sufficient since fn is passed.

Reply:
(see next comment)